specs/agents/tracing-spans-destination.md

```diff
@@ -0,0 +1,78 @@
+## Span destination
+
+Spans representing an external call MUST have `context.destination` information.
+If the span represents a call to an in-memory database, the information SHOULD still be set.
+
+Agents SHOULD have a generic component used in all tests that validates that the destination information is present for exit spans.
+Rather than opting into the validation, the testing should provide an opt-out if,
+for whatever reason, the destination information can't or shouldn't be collected for a particular exit span.
```

Comment:
I think all this should be said about `context.destination.service` fields only.
`context.destination.address` and `context.destination.port` have lower importance for us and they are much more likely to not be present for lots of exit span types, so I'd say agents `SHOULD` add them where readily available. For example, where the underlying protocol-layer-code is not readily available in the instrumented code, or when the instrumentation captures the exit event, but the actual client is not bound to a specific connection (e.g. a client that allows load balancing). In most cases we can probably get the address and port if needed, but the value would not justify the effort.